lib/resources/elasticsearch.rb

```diff
@@ -60,42 +60,51 @@ class Elasticsearch < Inspec.resource(1)
         its('name') { should include 'my_node' }
       end
       describe elasticsearch.nodes do
-          its('name') { should_not include 'node_name' }
+        its('name') { should_not include 'node_name' }
       end
       elasticsearch.nodes.os.each do |nodeos|
-      	describe nodeos do
-      		its('name'){ should_not cmp 'MacOS' }
-      	end
+        describe nodeos do
+          its('name'){ should_not cmp 'MacOS' }
+        end
       end
       elasticsearch.nodes.version.each do |node_version|
-      	describe node_version do
-      		it{ should be > '1.2.0' }
-      	end
+        describe node_version do
+          it{ should be > '1.2.0' }
+        end
       end
       elasticsearch.nodes.where{os.name == 'Linux' }.settings.each do |node_settings|
         describe node_settings do
           its('path.conf') {should cmp '/etc/elasticsearch'}
         end
       end
-		  elasticsearch.nodes.where{name == 'package-centos-72' }.settings.each do |node_settings|
+      elasticsearch.nodes.where{name == 'package-centos-72' }.settings.each do |node_settings|
         describe node_settings do
           its('path.conf') {should cmp '/etc/elasticsearch'}
         end
       end
         "
 
     def initialize(url = 'http://localhost:9200/_nodes/')
-      return skip_resource 'Package `curl` not avaiable on the host' unless inspec.package('curl').installed?
-      command = inspec.command("curl #{url}")
-      if command.stderr =~ /Failed to connect/
-        return skip_resource 'Connection refused please check ip and port provided'
-      end
-      @content = JSON.parse(command.stdout)
+      return skip_resource 'Package `curl` not avaiable on the host' unless inspec.command('curl').exist?
+      @content = read_es_data(url)
       return skip_resource 'Invalid node provided' unless !@content['_nodes']['successful'].zero?
       @nodes = []
       parse_cluster
     end
 
+    def read_es_data(url)
+      cmd = inspec.command("curl #{url}")
+      if !cmd.exit_status.zero?
+        return skip_resource "Error using the curl #{url}"
+      end
+      if cmd.stderr =~ /Failed to connect/
```

Comment:
If `curl` fails, it will catch at line 97 above and then we'll return. I don't think you'll ever get to this part of your method if it failed to connect.

If you want to provide a specific "connection refused" skip message, you will likely need to move lines 100-102 above line 97, and let the `if` statement at line 97 be a "catch-all error except connection refused events" handler. Make sense?